tests/integration/pipeline/test_bootstrap_command.py

```diff
@@ -0,0 +1,227 @@
+            self.assertNotIn("arn:aws:ecr:", stdout)
+            self.assertSetEqual(common_resources, self._extract_created_resource_logical_ids(stack_name))
+
+    @parameterized.expand([("create_ecr_repo",), (False,)])
+    def test_non_interactive_with_no_resources_provided(self, create_ecr_repo: bool):
+        stage_name, stack_name = self._get_stage_and_stack_name()
+        self.stack_names = [stack_name]
+
+        bootstrap_command_list = self.get_bootstrap_command_list(
+            no_interactive=True, create_ecr_repo=create_ecr_repo, no_confirm_changeset=True
+        )
+
+        bootstrap_process_execute = run_command(bootstrap_command_list)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 2)
+        stderr = bootstrap_process_execute.stderr.decode()
+        self.assertIn("Missing required parameter", stderr)
+
+    def test_interactive_with_all_required_resources_provided(self):
+        stage_name, stack_name = self._get_stage_and_stack_name()
+        self.stack_names = [stack_name]
+
+        bootstrap_command_list = self.get_bootstrap_command_list()
+
+        inputs = [
+            stage_name,
+            "arn:aws:iam::123:user/user-name",  # pipeline user
+            "arn:aws:iam::123:role/role-name",  # Pipeline execution role
+            "arn:aws:iam::123:role/role-name",  # CloudFormation execution role
+            "arn:aws:s3:::bucket-name",  # Artifacts bucket
+            "3",  # Should we create ECR repo, 3 - specify one
+            "arn:aws:ecr:::repository/repo-name",  # ecr repo
+            "1.2.3.4/24",  # Pipeline IP address range
+            "y",  # proceed
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertIn("skipping creation", stdout)
+
+    @parameterized.expand([("confirm_changeset",), (False,)])
+    def test_no_interactive_with_all_required_resources_provided(self, confirm_changeset):
+        stage_name, stack_name = self._get_stage_and_stack_name()
+        self.stack_names = [stack_name]
+
+        bootstrap_command_list = self.get_bootstrap_command_list(
+            no_interactive=True,
+            stage_name=stage_name,
+            pipeline_user="arn:aws:iam::123:user/user-name",  # pipeline user
+            pipeline_execution_role="arn:aws:iam::123:role/role-name",  # Pipeline execution role
+            cloudformation_execution_role="arn:aws:iam::123:role/role-name",  # CloudFormation execution role
+            artifacts_bucket="arn:aws:s3:::bucket-name",  # Artifacts bucket
+            ecr_repo="arn:aws:ecr:::repository/repo-name",  # ecr repo
+            pipeline_ip_range="1.2.3.4/24",  # Pipeline IP address range
+            no_confirm_changeset=confirm_changeset,
+        )
+
+        inputs = [
+            "y",  # proceed
+        ]
```

Comment:
can we make this something like `inputs = ["y"] if confirm_changeset else []` to make sure we don't pass "y" when there is no confrimation required.

Reply:
no worries, it is handled right below.